sapl/sessao/views.py

```diff
@@ -529,6 +529,168 @@ def get_presencas_generic(model, sessao, legislatura):
+            return context
+
+        if self.expediente:
+            context["title"] = self.title
+            context['lista_disponiveis'] = ExpedienteMateria.objects.filter(
+                sessao_plenaria=sessao_plenaria_atual
+            )
+
+        elif self.ordem:
+            context["title"] = self.title
+            context['lista_disponiveis'] = OrdemDia.objects.filter(
+                sessao_plenaria=sessao_plenaria_atual
+            )
+
+        context["numero_resultados"] = context['lista_disponiveis'].count()
+
+        if context['numero_resultados']:
+            context['sessoes'] = SessaoPlenaria.objects.filter(
+                data_inicio__gte=sessao_plenaria_atual.data_inicio
+            ).exclude(pk=sessao_plenaria_atual.pk).order_by("-data_inicio")
+        
+        return context
+
+    def post(self, request, *args, **kwargs):
+        marcadas = request.POST.getlist('opcao_id')
+
+        if not marcadas:
+            msg = _('Nenhuma matéria foi selecionada.')
+            messages.add_message(request, messages.ERROR, msg)
+            return self.get(request, self.kwargs)
+
+        sessao_plenaria_destino_id = request.POST['sessao_plenaria']
+        if not sessao_plenaria_destino_id:
+            self.logger.error(
+                "A variável sessao_plenaria da requisição de cópia de " \
+                "matérias entre sessões plenárias não existe."
+            )
+
+            msg = _('Ocorreu um erro inesperado. Tente novamente.')
+            messages.add_message(request, messages.ERROR, msg)
+
+            msg_c = _(
+                'Se o problema persistir, entre em contato com o suporte do ' \
+                'Interlegis.'
+            )
+            messages.add_message(request, messages.WARNING, msg_c)
+            
+            return self.get(request, self.kwargs)
+        
+        sessao = SessaoPlenaria.objects.get(id=sessao_plenaria_destino_id)
+        if self.expediente:
+            lista_expediente = []
+
+            numero_ordem = ExpedienteMateria.objects.filter(
+                sessao_plenaria=sessao
+            ).last().numero_ordem if ExpedienteMateria.objects.filter(
+                sessao_plenaria=sessao
+            ).exists() else 0
+
+            for num_ordem, expediente in enumerate(
+                ExpedienteMateria.objects.filter(id__in=marcadas),
+                numero_ordem+1
+            ):
+                lista_expediente.append(
+                    ExpedienteMateria(
+                        sessao_plenaria=sessao, materia=expediente.materia,
+                        data_ordem=expediente.data_ordem,
+                        observacao=expediente.observacao,
+                        numero_ordem=num_ordem,
+                        tipo_votacao=expediente.tipo_votacao,
+                        votacao_aberta=False, registro_aberto=False
+                    )
+                )
+            ExpedienteMateria.objects.bulk_create(lista_expediente)
```

Comment:
Imagine o seguinte cenário: a pessoa faz uma movimentação da sessão A para a sessão B um dia antes da sessão B de 50 matérias. No dia da sessão ela, na pressa, se esquece que já tinha copiado e copia de novo. Agora são 100 matérias, não?

Se for o caso, eu acho que deveria existir algum código assim pra expediente e ordem do dia:
```
materias_destino = recupera_materias(sessao_destino, 'expediente')
lista_expediente = [e for e in lista_expediente if e.materia not in materias_destino]
if lista_expediente:
    ExpedienteMateria.objects.bulk_create(lista_expediente)
```
lista_

Reply:
Solução implementada.